In [551]:
# Подключаем библиотеки
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

**1. Исходные данные**

In [552]:
df = pd.read_csv('/content/drive/MyDrive/t3_non_organic.csv', sep = ',')
df.head()

Attributed Touch Type Attributed Touch Time         Install Time  \
0                 click   2020-08-14 12:14:22  2020-08-14 13:09:32   
1                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
2                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
3                 click   2020-08-15 01:34:25  2020-08-15 01:37:29   
4                 click   2020-08-15 14:03:15  2020-08-15 14:21:59   

            Event Time       Event Name  \
0  2020-08-15 23:59:53  af_content_view   
1  2020-08-15 23:59:47  af_content_view   
2  2020-08-15 23:59:43  af_content_view   
3  2020-08-15 23:59:42  af_content_view   
4  2020-08-15 23:59:42  af_content_view   

                                         Event Value  Event Revenue  \
0    {"af_content":"","af_content_type":"user-chat"}            NaN   
1  {"af_content":"point:40","af_content_type":"qu...            NaN   
2  {"af_content":"step:2","af_content_type":"mini...            NaN   
3  {"af_content":"mode:1vs1Rank","af_content_type...            NaN   
4  {"af_content":"name:Imp level:2","af_content_t...            NaN   

  Event Revenue Currency  Event Revenue USD       Media Source  ... Language  \
0                    USD                NaN       Facebook Ads  ...  русский   
1                    USD                NaN       Facebook Ads  ...  русский   
2                    USD                NaN       Facebook Ads  ...  русский   
3                    USD                NaN  googleadwords_int  ...  русский   
4                    USD                NaN       Facebook Ads  ...  English   

                        AppsFlyer ID                        Advertising ID  \
0  1597410570684-3799928820535696190  2bca3e9d-1c06-4c13-8895-c311f9d8bfdd   
1  1597420480729-1759484924098917478  a9423486-6065-42ac-9b75-54c712da3435   
2  1597420480729-1759484924098917478  a9423486-6065-42ac-9b75-54c712da3435   
3  1597455446369-7748387263592684895  96acf210-19f7-4613-9259-8af44c5744ed   
4  1597501316454-1211890571372415119  48edfbd9-212a-477b-884e-46215c7ec482   

  IDFA  Platform             Device Type  OS Version Attribution Lookback  \
0  NaN   android     xiaomi-Redmi Note 5           9                  28d   
1  NaN   android  Redmi-Redmi Note 9 Pro          10                  28d   
2  NaN   android  Redmi-Redmi Note 9 Pro          10                  28d   
3  NaN   android         samsung-SM-T515           9                  30d   
4  NaN   android            nubia-NX659J          10                  28d   

  Is Primary Attribution                                         User Agent  
0                   True  Dalvik/2.1.0 (Linux; U; Android 9; Redmi Note ...  
1                   True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...  
2                   True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...  
3                   True  Dalvik/2.1.0 (Linux; U; Android 9; SM-T515 Bui...  
4                   True  Dalvik/2.1.0 (Linux; U; Android 10; NX659J Bui...  

[5 rows x 33 columns]

**Первичный анализ данных. Подготовка данных.**

In [553]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 59603 entries, 0 to 59602
Data columns (total 33 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Attributed Touch Type   57612 non-null  object 
 1   Attributed Touch Time   57612 non-null  object 
 2   Install Time            59603 non-null  object 
 3   Event Time              59603 non-null  object 
 4   Event Name              59603 non-null  object 
 5   Event Value             59603 non-null  object 
 6   Event Revenue           7 non-null      float64
 7   Event Revenue Currency  59603 non-null  object 
 8   Event Revenue USD       7 non-null      float64
 9   Media Source            59603 non-null  object 
 10  Channel                 57612 non-null  object 
 11  Campaign                57612 non-null  object 
 12  Campaign ID             57612 non-null  float64
 13  Adset                   51975 non-null  object 
 14  Adset ID                51975 non-null

In [554]:
print(df['Event Revenue'].unique())

[   nan 219.99  10.99   8.29 279.99  12.99]


In [555]:
df.shape

(59603, 33)

**2. Основные задачи**

**2.1.** Собрать когортный отчет, где за когорту взять весь трафик (когорта от установки) в разбивке по стране.<br>
В качестве визуализации составить табличку в Google Sheets, в строках которой будут страны, а в столбцах набор метрик:<br>
a. Installs<br>
b. Payers - число платящих<br>
c. Purchases - число покупок<br>
d. Tutorial Finished - процент прохождения тутора<br>
e. level_10% - процент прохождения уровня<br>
f. C2p - процент платящих<br>
g. RPI - доход на одну установку<br>
h. Revenue - доход в абсолютном значении


In [556]:
print(df.columns.values)

['Attributed Touch Type' 'Attributed Touch Time' 'Install Time'
 'Event Time' 'Event Name' 'Event Value' 'Event Revenue'
 'Event Revenue Currency' 'Event Revenue USD' 'Media Source' 'Channel'
 'Campaign' 'Campaign ID' 'Adset' 'Adset ID' 'Ad' 'Ad ID' 'Ad Type'
 'Site ID' 'Region' 'Country Code' 'IP' 'WIFI' 'Language' 'AppsFlyer ID'
 'Advertising ID' 'IDFA' 'Platform' 'Device Type' 'OS Version'
 'Attribution Lookback' 'Is Primary Attribution' 'User Agent']


In [557]:
df['Event Name'].unique()

array(['af_content_view', 'LaunchApp', 'FirstLaunchApp',
       'af_tutorial_completion', 'af_purchase', 'achieve_level_10',
       'purchase_ta_starter', 'achieve_level_15',
       'purchase_ta_challenge_pass'], dtype=object)

In [558]:
df.sort_values('Country Code')
df.head()

Attributed Touch Type Attributed Touch Time         Install Time  \
0                 click   2020-08-14 12:14:22  2020-08-14 13:09:32   
1                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
2                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
3                 click   2020-08-15 01:34:25  2020-08-15 01:37:29   
4                 click   2020-08-15 14:03:15  2020-08-15 14:21:59   

            Event Time       Event Name  \
0  2020-08-15 23:59:53  af_content_view   
1  2020-08-15 23:59:47  af_content_view   
2  2020-08-15 23:59:43  af_content_view   
3  2020-08-15 23:59:42  af_content_view   
4  2020-08-15 23:59:42  af_content_view   

                                         Event Value  Event Revenue  \
0    {"af_content":"","af_content_type":"user-chat"}            NaN   
1  {"af_content":"point:40","af_content_type":"qu...            NaN   
2  {"af_content":"step:2","af_content_type":"mini...            NaN   
3  {"af_content":"mode:1vs1Rank","af_content_type...            NaN   
4  {"af_content":"name:Imp level:2","af_content_t...            NaN   

  Event Revenue Currency  Event Revenue USD       Media Source  ... Language  \
0                    USD                NaN       Facebook Ads  ...  русский   
1                    USD                NaN       Facebook Ads  ...  русский   
2                    USD                NaN       Facebook Ads  ...  русский   
3                    USD                NaN  googleadwords_int  ...  русский   
4                    USD                NaN       Facebook Ads  ...  English   

                        AppsFlyer ID                        Advertising ID  \
0  1597410570684-3799928820535696190  2bca3e9d-1c06-4c13-8895-c311f9d8bfdd   
1  1597420480729-1759484924098917478  a9423486-6065-42ac-9b75-54c712da3435   
2  1597420480729-1759484924098917478  a9423486-6065-42ac-9b75-54c712da3435   
3  1597455446369-7748387263592684895  96acf210-19f7-4613-9259-8af44c5744ed   
4  1597501316454-1211890571372415119  48edfbd9-212a-477b-884e-46215c7ec482   

  IDFA  Platform             Device Type  OS Version Attribution Lookback  \
0  NaN   android     xiaomi-Redmi Note 5           9                  28d   
1  NaN   android  Redmi-Redmi Note 9 Pro          10                  28d   
2  NaN   android  Redmi-Redmi Note 9 Pro          10                  28d   
3  NaN   android         samsung-SM-T515           9                  30d   
4  NaN   android            nubia-NX659J          10                  28d   

  Is Primary Attribution                                         User Agent  
0                   True  Dalvik/2.1.0 (Linux; U; Android 9; Redmi Note ...  
1                   True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...  
2                   True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...  
3                   True  Dalvik/2.1.0 (Linux; U; Android 9; SM-T515 Bui...  
4                   True  Dalvik/2.1.0 (Linux; U; Android 10; NX659J Bui...  

[5 rows x 33 columns]

In [560]:
df[df['Event Revenue'].isna() == True]
df.head()

Attributed Touch Type Attributed Touch Time         Install Time  \
0                 click   2020-08-14 12:14:22  2020-08-14 13:09:32   
1                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
2                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
3                 click   2020-08-15 01:34:25  2020-08-15 01:37:29   
4                 click   2020-08-15 14:03:15  2020-08-15 14:21:59   

            Event Time       Event Name  \
0  2020-08-15 23:59:53  af_content_view   
1  2020-08-15 23:59:47  af_content_view   
2  2020-08-15 23:59:43  af_content_view   
3  2020-08-15 23:59:42  af_content_view   
4  2020-08-15 23:59:42  af_content_view   

                                         Event Value  Event Revenue  \
0    {"af_content":"","af_content_type":"user-chat"}            NaN   
1  {"af_content":"point:40","af_content_type":"qu...            NaN   
2  {"af_content":"step:2","af_content_type":"mini...            NaN   
3  {"af_content":"mode:1vs1Rank","af_content_type...            NaN   
4  {"af_content":"name:Imp level:2","af_content_t...            NaN   

  Event Revenue Currency  Event Revenue USD       Media Source  ... Language  \
0                    USD                NaN       Facebook Ads  ...  русский   
1                    USD                NaN       Facebook Ads  ...  русский   
2                    USD                NaN       Facebook Ads  ...  русский   
3                    USD                NaN  googleadwords_int  ...  русский   
4                    USD                NaN       Facebook Ads  ...  English   

                        AppsFlyer ID                        Advertising ID  \
0  1597410570684-3799928820535696190  2bca3e9d-1c06-4c13-8895-c311f9d8bfdd   
1  1597420480729-1759484924098917478  a9423486-6065-42ac-9b75-54c712da3435   
2  1597420480729-1759484924098917478  a9423486-6065-42ac-9b75-54c712da3435   
3  1597455446369-7748387263592684895  96acf210-19f7-4613-9259-8af44c5744ed   
4  1597501316454-1211890571372415119  48edfbd9-212a-477b-884e-46215c7ec482   

  IDFA  Platform             Device Type  OS Version Attribution Lookback  \
0  NaN   android     xiaomi-Redmi Note 5           9                  28d   
1  NaN   android  Redmi-Redmi Note 9 Pro          10                  28d   
2  NaN   android  Redmi-Redmi Note 9 Pro          10                  28d   
3  NaN   android         samsung-SM-T515           9                  30d   
4  NaN   android            nubia-NX659J          10                  28d   

  Is Primary Attribution                                         User Agent  
0                   True  Dalvik/2.1.0 (Linux; U; Android 9; Redmi Note ...  
1                   True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...  
2                   True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...  
3                   True  Dalvik/2.1.0 (Linux; U; Android 9; SM-T515 Bui...  
4                   True  Dalvik/2.1.0 (Linux; U; Android 10; NX659J Bui...  

[5 rows x 33 columns]

In [561]:
df[df['IDFA'].isna() == True]
df.head()

Attributed Touch Type Attributed Touch Time         Install Time  \
0                 click   2020-08-14 12:14:22  2020-08-14 13:09:32   
1                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
2                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
3                 click   2020-08-15 01:34:25  2020-08-15 01:37:29   
4                 click   2020-08-15 14:03:15  2020-08-15 14:21:59   

            Event Time       Event Name  \
0  2020-08-15 23:59:53  af_content_view   
1  2020-08-15 23:59:47  af_content_view   
2  2020-08-15 23:59:43  af_content_view   
3  2020-08-15 23:59:42  af_content_view   
4  2020-08-15 23:59:42  af_content_view   

                                         Event Value  Event Revenue  \
0    {"af_content":"","af_content_type":"user-chat"}            NaN   
1  {"af_content":"point:40","af_content_type":"qu...            NaN   
2  {"af_content":"step:2","af_content_type":"mini...            NaN   
3  {"af_content":"mode:1vs1Rank","af_content_type...            NaN   
4  {"af_content":"name:Imp level:2","af_content_t...            NaN   

  Event Revenue Currency  Event Revenue USD       Media Source  ... Language  \
0                    USD                NaN       Facebook Ads  ...  русский   
1                    USD                NaN       Facebook Ads  ...  русский   
2                    USD                NaN       Facebook Ads  ...  русский   
3                    USD                NaN  googleadwords_int  ...  русский   
4                    USD                NaN       Facebook Ads  ...  English   

                        AppsFlyer ID                        Advertising ID  \
0  1597410570684-3799928820535696190  2bca3e9d-1c06-4c13-8895-c311f9d8bfdd   
1  1597420480729-1759484924098917478  a9423486-6065-42ac-9b75-54c712da3435   
2  1597420480729-1759484924098917478  a9423486-6065-42ac-9b75-54c712da3435   
3  1597455446369-7748387263592684895  96acf210-19f7-4613-9259-8af44c5744ed   
4  1597501316454-1211890571372415119  48edfbd9-212a-477b-884e-46215c7ec482   

  IDFA  Platform             Device Type  OS Version Attribution Lookback  \
0  NaN   android     xiaomi-Redmi Note 5           9                  28d   
1  NaN   android  Redmi-Redmi Note 9 Pro          10                  28d   
2  NaN   android  Redmi-Redmi Note 9 Pro          10                  28d   
3  NaN   android         samsung-SM-T515           9                  30d   
4  NaN   android            nubia-NX659J          10                  28d   

  Is Primary Attribution                                         User Agent  
0                   True  Dalvik/2.1.0 (Linux; U; Android 9; Redmi Note ...  
1                   True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...  
2                   True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...  
3                   True  Dalvik/2.1.0 (Linux; U; Android 9; SM-T515 Bui...  
4                   True  Dalvik/2.1.0 (Linux; U; Android 10; NX659J Bui...  

[5 rows x 33 columns]

In [562]:
df['Installs'] = df.groupby('Country Code')['Install Time'].transform('count')
df.head()

Attributed Touch Type Attributed Touch Time         Install Time  \
0                 click   2020-08-14 12:14:22  2020-08-14 13:09:32   
1                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
2                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
3                 click   2020-08-15 01:34:25  2020-08-15 01:37:29   
4                 click   2020-08-15 14:03:15  2020-08-15 14:21:59   

            Event Time       Event Name  \
0  2020-08-15 23:59:53  af_content_view   
1  2020-08-15 23:59:47  af_content_view   
2  2020-08-15 23:59:43  af_content_view   
3  2020-08-15 23:59:42  af_content_view   
4  2020-08-15 23:59:42  af_content_view   

                                         Event Value  Event Revenue  \
0    {"af_content":"","af_content_type":"user-chat"}            NaN   
1  {"af_content":"point:40","af_content_type":"qu...            NaN   
2  {"af_content":"step:2","af_content_type":"mini...            NaN   
3  {"af_content":"mode:1vs1Rank","af_content_type...            NaN   
4  {"af_content":"name:Imp level:2","af_content_t...            NaN   

  Event Revenue Currency  Event Revenue USD       Media Source  ...  \
0                    USD                NaN       Facebook Ads  ...   
1                    USD                NaN       Facebook Ads  ...   
2                    USD                NaN       Facebook Ads  ...   
3                    USD                NaN  googleadwords_int  ...   
4                    USD                NaN       Facebook Ads  ...   

                        AppsFlyer ID                        Advertising ID  \
0  1597410570684-3799928820535696190  2bca3e9d-1c06-4c13-8895-c311f9d8bfdd   
1  1597420480729-1759484924098917478  a9423486-6065-42ac-9b75-54c712da3435   
2  1597420480729-1759484924098917478  a9423486-6065-42ac-9b75-54c712da3435   
3  1597455446369-7748387263592684895  96acf210-19f7-4613-9259-8af44c5744ed   
4  1597501316454-1211890571372415119  48edfbd9-212a-477b-884e-46215c7ec482   

   IDFA Platform             Device Type OS Version  Attribution Lookback  \
0   NaN  android     xiaomi-Redmi Note 5          9                   28d   
1   NaN  android  Redmi-Redmi Note 9 Pro         10                   28d   
2   NaN  android  Redmi-Redmi Note 9 Pro         10                   28d   
3   NaN  android         samsung-SM-T515          9                   30d   
4   NaN  android            nubia-NX659J         10                   28d   

  Is Primary Attribution                                         User Agent  \
0                   True  Dalvik/2.1.0 (Linux; U; Android 9; Redmi Note ...   
1                   True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...   
2                   True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...   
3                   True  Dalvik/2.1.0 (Linux; U; Android 9; SM-T515 Bui...   
4                   True  Dalvik/2.1.0 (Linux; U; Android 10; NX659J Bui...   

  Installs  
0    29556  
1     7787  
2     7787  
3    29556  
4     4655  

[5 rows x 34 columns]

In [563]:
df['Payers'] = df.groupby('Country Code')['Event Revenue'].transform('count')
df.head()

Attributed Touch Type Attributed Touch Time         Install Time  \
0                 click   2020-08-14 12:14:22  2020-08-14 13:09:32   
1                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
2                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
3                 click   2020-08-15 01:34:25  2020-08-15 01:37:29   
4                 click   2020-08-15 14:03:15  2020-08-15 14:21:59   

            Event Time       Event Name  \
0  2020-08-15 23:59:53  af_content_view   
1  2020-08-15 23:59:47  af_content_view   
2  2020-08-15 23:59:43  af_content_view   
3  2020-08-15 23:59:42  af_content_view   
4  2020-08-15 23:59:42  af_content_view   

                                         Event Value  Event Revenue  \
0    {"af_content":"","af_content_type":"user-chat"}            NaN   
1  {"af_content":"point:40","af_content_type":"qu...            NaN   
2  {"af_content":"step:2","af_content_type":"mini...            NaN   
3  {"af_content":"mode:1vs1Rank","af_content_type...            NaN   
4  {"af_content":"name:Imp level:2","af_content_t...            NaN   

  Event Revenue Currency  Event Revenue USD       Media Source  ...  \
0                    USD                NaN       Facebook Ads  ...   
1                    USD                NaN       Facebook Ads  ...   
2                    USD                NaN       Facebook Ads  ...   
3                    USD                NaN  googleadwords_int  ...   
4                    USD                NaN       Facebook Ads  ...   

                         Advertising ID IDFA  Platform  \
0  2bca3e9d-1c06-4c13-8895-c311f9d8bfdd  NaN   android   
1  a9423486-6065-42ac-9b75-54c712da3435  NaN   android   
2  a9423486-6065-42ac-9b75-54c712da3435  NaN   android   
3  96acf210-19f7-4613-9259-8af44c5744ed  NaN   android   
4  48edfbd9-212a-477b-884e-46215c7ec482  NaN   android   

              Device Type  OS Version Attribution Lookback  \
0     xiaomi-Redmi Note 5           9                  28d   
1  Redmi-Redmi Note 9 Pro          10                  28d   
2  Redmi-Redmi Note 9 Pro          10                  28d   
3         samsung-SM-T515           9                  30d   
4            nubia-NX659J          10                  28d   

   Is Primary Attribution                                         User Agent  \
0                    True  Dalvik/2.1.0 (Linux; U; Android 9; Redmi Note ...   
1                    True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...   
2                    True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...   
3                    True  Dalvik/2.1.0 (Linux; U; Android 9; SM-T515 Bui...   
4                    True  Dalvik/2.1.0 (Linux; U; Android 10; NX659J Bui...   

  Installs Payers  
0    29556      3  
1     7787      1  
2     7787      1  
3    29556      3  
4     4655      3  

[5 rows x 35 columns]

In [564]:
df['Purchases'] = df.groupby('Country Code')['Event Revenue'].transform('count')
df.head()

Attributed Touch Type Attributed Touch Time         Install Time  \
0                 click   2020-08-14 12:14:22  2020-08-14 13:09:32   
1                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
2                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
3                 click   2020-08-15 01:34:25  2020-08-15 01:37:29   
4                 click   2020-08-15 14:03:15  2020-08-15 14:21:59   

            Event Time       Event Name  \
0  2020-08-15 23:59:53  af_content_view   
1  2020-08-15 23:59:47  af_content_view   
2  2020-08-15 23:59:43  af_content_view   
3  2020-08-15 23:59:42  af_content_view   
4  2020-08-15 23:59:42  af_content_view   

                                         Event Value  Event Revenue  \
0    {"af_content":"","af_content_type":"user-chat"}            NaN   
1  {"af_content":"point:40","af_content_type":"qu...            NaN   
2  {"af_content":"step:2","af_content_type":"mini...            NaN   
3  {"af_content":"mode:1vs1Rank","af_content_type...            NaN   
4  {"af_content":"name:Imp level:2","af_content_t...            NaN   

  Event Revenue Currency  Event Revenue USD       Media Source  ... IDFA  \
0                    USD                NaN       Facebook Ads  ...  NaN   
1                    USD                NaN       Facebook Ads  ...  NaN   
2                    USD                NaN       Facebook Ads  ...  NaN   
3                    USD                NaN  googleadwords_int  ...  NaN   
4                    USD                NaN       Facebook Ads  ...  NaN   

  Platform             Device Type OS Version  Attribution Lookback  \
0  android     xiaomi-Redmi Note 5          9                   28d   
1  android  Redmi-Redmi Note 9 Pro         10                   28d   
2  android  Redmi-Redmi Note 9 Pro         10                   28d   
3  android         samsung-SM-T515          9                   30d   
4  android            nubia-NX659J         10                   28d   

  Is Primary Attribution                                         User Agent  \
0                   True  Dalvik/2.1.0 (Linux; U; Android 9; Redmi Note ...   
1                   True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...   
2                   True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...   
3                   True  Dalvik/2.1.0 (Linux; U; Android 9; SM-T515 Bui...   
4                   True  Dalvik/2.1.0 (Linux; U; Android 10; NX659J Bui...   

  Installs Payers Purchases  
0    29556      3         3  
1     7787      1         1  
2     7787      1         1  
3    29556      3         3  
4     4655      3         3  

[5 rows x 36 columns]

In [565]:
df['Revenue'] =round((df.groupby('Country Code')['Event Revenue USD'].transform('sum')), 2)
df.head()

Attributed Touch Type Attributed Touch Time         Install Time  \
0                 click   2020-08-14 12:14:22  2020-08-14 13:09:32   
1                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
2                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
3                 click   2020-08-15 01:34:25  2020-08-15 01:37:29   
4                 click   2020-08-15 14:03:15  2020-08-15 14:21:59   

            Event Time       Event Name  \
0  2020-08-15 23:59:53  af_content_view   
1  2020-08-15 23:59:47  af_content_view   
2  2020-08-15 23:59:43  af_content_view   
3  2020-08-15 23:59:42  af_content_view   
4  2020-08-15 23:59:42  af_content_view   

                                         Event Value  Event Revenue  \
0    {"af_content":"","af_content_type":"user-chat"}            NaN   
1  {"af_content":"point:40","af_content_type":"qu...            NaN   
2  {"af_content":"step:2","af_content_type":"mini...            NaN   
3  {"af_content":"mode:1vs1Rank","af_content_type...            NaN   
4  {"af_content":"name:Imp level:2","af_content_t...            NaN   

  Event Revenue Currency  Event Revenue USD       Media Source  ... Platform  \
0                    USD                NaN       Facebook Ads  ...  android   
1                    USD                NaN       Facebook Ads  ...  android   
2                    USD                NaN       Facebook Ads  ...  android   
3                    USD                NaN  googleadwords_int  ...  android   
4                    USD                NaN       Facebook Ads  ...  android   

              Device Type  OS Version Attribution Lookback  \
0     xiaomi-Redmi Note 5           9                  28d   
1  Redmi-Redmi Note 9 Pro          10                  28d   
2  Redmi-Redmi Note 9 Pro          10                  28d   
3         samsung-SM-T515           9                  30d   
4            nubia-NX659J          10                  28d   

   Is Primary Attribution                                         User Agent  \
0                    True  Dalvik/2.1.0 (Linux; U; Android 9; Redmi Note ...   
1                    True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...   
2                    True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...   
3                    True  Dalvik/2.1.0 (Linux; U; Android 9; SM-T515 Bui...   
4                    True  Dalvik/2.1.0 (Linux; U; Android 10; NX659J Bui...   

   Installs Payers Purchases Revenue  
0     29556      3         3   26.34  
1      7787      1         1    8.29  
2      7787      1         1    8.29  
3     29556      3         3   26.34  
4      4655      3         3   26.40  

[5 rows x 37 columns]

In [566]:
df['RPI'] = round((df['Revenue']/df['Installs']),4)
df.head()

Attributed Touch Type Attributed Touch Time         Install Time  \
0                 click   2020-08-14 12:14:22  2020-08-14 13:09:32   
1                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
2                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
3                 click   2020-08-15 01:34:25  2020-08-15 01:37:29   
4                 click   2020-08-15 14:03:15  2020-08-15 14:21:59   

            Event Time       Event Name  \
0  2020-08-15 23:59:53  af_content_view   
1  2020-08-15 23:59:47  af_content_view   
2  2020-08-15 23:59:43  af_content_view   
3  2020-08-15 23:59:42  af_content_view   
4  2020-08-15 23:59:42  af_content_view   

                                         Event Value  Event Revenue  \
0    {"af_content":"","af_content_type":"user-chat"}            NaN   
1  {"af_content":"point:40","af_content_type":"qu...            NaN   
2  {"af_content":"step:2","af_content_type":"mini...            NaN   
3  {"af_content":"mode:1vs1Rank","af_content_type...            NaN   
4  {"af_content":"name:Imp level:2","af_content_t...            NaN   

  Event Revenue Currency  Event Revenue USD       Media Source  ...  \
0                    USD                NaN       Facebook Ads  ...   
1                    USD                NaN       Facebook Ads  ...   
2                    USD                NaN       Facebook Ads  ...   
3                    USD                NaN  googleadwords_int  ...   
4                    USD                NaN       Facebook Ads  ...   

              Device Type OS Version  Attribution Lookback  \
0     xiaomi-Redmi Note 5          9                   28d   
1  Redmi-Redmi Note 9 Pro         10                   28d   
2  Redmi-Redmi Note 9 Pro         10                   28d   
3         samsung-SM-T515          9                   30d   
4            nubia-NX659J         10                   28d   

  Is Primary Attribution                                         User Agent  \
0                   True  Dalvik/2.1.0 (Linux; U; Android 9; Redmi Note ...   
1                   True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...   
2                   True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...   
3                   True  Dalvik/2.1.0 (Linux; U; Android 9; SM-T515 Bui...   
4                   True  Dalvik/2.1.0 (Linux; U; Android 10; NX659J Bui...   

  Installs  Payers Purchases Revenue     RPI  
0    29556       3         3   26.34  0.0009  
1     7787       1         1    8.29  0.0011  
2     7787       1         1    8.29  0.0011  
3    29556       3         3   26.34  0.0009  
4     4655       3         3   26.40  0.0057  

[5 rows x 38 columns]

In [567]:
df['C2p'] = round(((df['Payers']/ df['Installs'])*100),4)
df.head()

Attributed Touch Type Attributed Touch Time         Install Time  \
0                 click   2020-08-14 12:14:22  2020-08-14 13:09:32   
1                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
2                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
3                 click   2020-08-15 01:34:25  2020-08-15 01:37:29   
4                 click   2020-08-15 14:03:15  2020-08-15 14:21:59   

            Event Time       Event Name  \
0  2020-08-15 23:59:53  af_content_view   
1  2020-08-15 23:59:47  af_content_view   
2  2020-08-15 23:59:43  af_content_view   
3  2020-08-15 23:59:42  af_content_view   
4  2020-08-15 23:59:42  af_content_view   

                                         Event Value  Event Revenue  \
0    {"af_content":"","af_content_type":"user-chat"}            NaN   
1  {"af_content":"point:40","af_content_type":"qu...            NaN   
2  {"af_content":"step:2","af_content_type":"mini...            NaN   
3  {"af_content":"mode:1vs1Rank","af_content_type...            NaN   
4  {"af_content":"name:Imp level:2","af_content_t...            NaN   

  Event Revenue Currency  Event Revenue USD       Media Source  ...  \
0                    USD                NaN       Facebook Ads  ...   
1                    USD                NaN       Facebook Ads  ...   
2                    USD                NaN       Facebook Ads  ...   
3                    USD                NaN  googleadwords_int  ...   
4                    USD                NaN       Facebook Ads  ...   

  OS Version Attribution Lookback  Is Primary Attribution  \
0          9                  28d                    True   
1         10                  28d                    True   
2         10                  28d                    True   
3          9                  30d                    True   
4         10                  28d                    True   

                                          User Agent  Installs Payers  \
0  Dalvik/2.1.0 (Linux; U; Android 9; Redmi Note ...     29556      3   
1  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...      7787      1   
2  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...      7787      1   
3  Dalvik/2.1.0 (Linux; U; Android 9; SM-T515 Bui...     29556      3   
4  Dalvik/2.1.0 (Linux; U; Android 10; NX659J Bui...      4655      3   

   Purchases Revenue     RPI     C2p  
0          3   26.34  0.0009  0.0102  
1          1    8.29  0.0011  0.0128  
2          1    8.29  0.0011  0.0128  
3          3   26.34  0.0009  0.0102  
4          3   26.40  0.0057  0.0644  

[5 rows x 39 columns]

In [568]:
df['level_10%'] = round((((len(df[df['Event Name']=='achieve_level_10']))/df['Installs'])),0)
df.head()

Attributed Touch Type Attributed Touch Time         Install Time  \
0                 click   2020-08-14 12:14:22  2020-08-14 13:09:32   
1                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
2                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
3                 click   2020-08-15 01:34:25  2020-08-15 01:37:29   
4                 click   2020-08-15 14:03:15  2020-08-15 14:21:59   

            Event Time       Event Name  \
0  2020-08-15 23:59:53  af_content_view   
1  2020-08-15 23:59:47  af_content_view   
2  2020-08-15 23:59:43  af_content_view   
3  2020-08-15 23:59:42  af_content_view   
4  2020-08-15 23:59:42  af_content_view   

                                         Event Value  Event Revenue  \
0    {"af_content":"","af_content_type":"user-chat"}            NaN   
1  {"af_content":"point:40","af_content_type":"qu...            NaN   
2  {"af_content":"step:2","af_content_type":"mini...            NaN   
3  {"af_content":"mode:1vs1Rank","af_content_type...            NaN   
4  {"af_content":"name:Imp level:2","af_content_t...            NaN   

  Event Revenue Currency  Event Revenue USD       Media Source  ...  \
0                    USD                NaN       Facebook Ads  ...   
1                    USD                NaN       Facebook Ads  ...   
2                    USD                NaN       Facebook Ads  ...   
3                    USD                NaN  googleadwords_int  ...   
4                    USD                NaN       Facebook Ads  ...   

  Attribution Lookback Is Primary Attribution  \
0                  28d                   True   
1                  28d                   True   
2                  28d                   True   
3                  30d                   True   
4                  28d                   True   

                                          User Agent Installs  Payers  \
0  Dalvik/2.1.0 (Linux; U; Android 9; Redmi Note ...    29556       3   
1  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...     7787       1   
2  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...     7787       1   
3  Dalvik/2.1.0 (Linux; U; Android 9; SM-T515 Bui...    29556       3   
4  Dalvik/2.1.0 (Linux; U; Android 10; NX659J Bui...     4655       3   

  Purchases  Revenue     RPI     C2p level_10%  
0         3    26.34  0.0009  0.0102       0.0  
1         1     8.29  0.0011  0.0128       0.0  
2         1     8.29  0.0011  0.0128       0.0  
3         3    26.34  0.0009  0.0102       0.0  
4         3    26.40  0.0057  0.0644       0.0  

[5 rows x 40 columns]

In [569]:
df['Tutorial Finished'] = round((((len(df[df['Event Name']=='af_tutorial_completion'])*0.1)/df['Installs'])),0)
df.head()

Attributed Touch Type Attributed Touch Time         Install Time  \
0                 click   2020-08-14 12:14:22  2020-08-14 13:09:32   
1                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
2                 click   2020-08-12 19:41:19  2020-08-14 15:54:44   
3                 click   2020-08-15 01:34:25  2020-08-15 01:37:29   
4                 click   2020-08-15 14:03:15  2020-08-15 14:21:59   

            Event Time       Event Name  \
0  2020-08-15 23:59:53  af_content_view   
1  2020-08-15 23:59:47  af_content_view   
2  2020-08-15 23:59:43  af_content_view   
3  2020-08-15 23:59:42  af_content_view   
4  2020-08-15 23:59:42  af_content_view   

                                         Event Value  Event Revenue  \
0    {"af_content":"","af_content_type":"user-chat"}            NaN   
1  {"af_content":"point:40","af_content_type":"qu...            NaN   
2  {"af_content":"step:2","af_content_type":"mini...            NaN   
3  {"af_content":"mode:1vs1Rank","af_content_type...            NaN   
4  {"af_content":"name:Imp level:2","af_content_t...            NaN   

  Event Revenue Currency  Event Revenue USD       Media Source  ...  \
0                    USD                NaN       Facebook Ads  ...   
1                    USD                NaN       Facebook Ads  ...   
2                    USD                NaN       Facebook Ads  ...   
3                    USD                NaN  googleadwords_int  ...   
4                    USD                NaN       Facebook Ads  ...   

  Is Primary Attribution                                         User Agent  \
0                   True  Dalvik/2.1.0 (Linux; U; Android 9; Redmi Note ...   
1                   True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...   
2                   True  Dalvik/2.1.0 (Linux; U; Android 10; Redmi Note...   
3                   True  Dalvik/2.1.0 (Linux; U; Android 9; SM-T515 Bui...   
4                   True  Dalvik/2.1.0 (Linux; U; Android 10; NX659J Bui...   

   Installs Payers  Purchases Revenue     RPI     C2p level_10%  \
0     29556      3          3   26.34  0.0009  0.0102       0.0   
1      7787      1          1    8.29  0.0011  0.0128       0.0   
2      7787      1          1    8.29  0.0011  0.0128       0.0   
3     29556      3          3   26.34  0.0009  0.0102       0.0   
4      4655      3          3   26.40  0.0057  0.0644       0.0   

  Tutorial Finished  
0               0.0  
1               0.0  
2               0.0  
3               0.0  
4               0.0  

[5 rows x 41 columns]

In [ ]:
df = df.drop(['Ad ID', 'Adset ID', 'Campaign ID','Event Revenue', 'Event Revenue USD', 'Is Primary Attribution', 'WIFI'], axis=1)
df.head()

In [572]:
df.pivot_table(index='Country Code')

<ipython-input-572-6b1c071ec83b>:1: FutureWarning: pivot_table dropped a column because it failed to aggregate. This behavior is deprecated and will raise in a future version of pandas. Select only the columns that can be aggregated.
  df.pivot_table(index='Country Code')


C2p  Installs  Payers  Purchases     RPI  Revenue  \
Country Code                                                         
AF            0.0000         9       0          0  0.0000     0.00   
AM            0.0000       415       0          0  0.0000     0.00   
AZ            0.0000      1224       0          0  0.0000     0.00   
BG            0.0000         1       0          0  0.0000     0.00   
BY            0.0128      7787       1          1  0.0011     8.29   
CA            0.0644      4655       3          3  0.0057    26.40   
CZ            0.0000        22       0          0  0.0000     0.00   
DE            0.0000        35       0          0  0.0000     0.00   
DK            0.0000         1       0          0  0.0000     0.00   
ES            0.0000         3       0          0  0.0000     0.00   
FR            0.0000        27       0          0  0.0000     0.00   
GE            0.0000      2238       0          0  0.0000     0.00   
HU            0.0000        10       0          0  0.0000     0.00   
KG            0.0000      1640       0          0  0.0000     0.00   
KZ            0.0000      7504       0          0  0.0000     0.00   
MD            0.0000      1095       0          0  0.0000     0.00   
NL            0.0000        11       0          0  0.0000     0.00   
PL            0.0000       353       0          0  0.0000     0.00   
PT            0.0000         9       0          0  0.0000     0.00   
RU            0.0000       430       0          0  0.0000     0.00   
SK            0.0000        82       0          0  0.0000     0.00   
TJ            0.0000       202       0          0  0.0000     0.00   
TM            0.0000         2       0          0  0.0000     0.00   
UA            0.0102     29556       3          3  0.0009    26.34   
US            0.0000         4       0          0  0.0000     0.00   
UZ            0.0000      2288       0          0  0.0000     0.00   

              Tutorial Finished  level_10%  
Country Code                                
AF                          4.0        3.0  
AM                          0.0        0.0  
AZ                          0.0        0.0  
BG                         40.0       23.0  
BY                          0.0        0.0  
CA                          0.0        0.0  
CZ                          2.0        1.0  
DE                          1.0        1.0  
DK                         40.0       23.0  
ES                         13.0        8.0  
FR                          1.0        1.0  
GE                          0.0        0.0  
HU                          4.0        2.0  
KG                          0.0        0.0  
KZ                          0.0        0.0  
MD                          0.0        0.0  
NL                          4.0        2.0  
PL                          0.0        0.0  
PT                          4.0        3.0  
RU                          0.0        0.0  
SK                          0.0        0.0  
TJ                          0.0        0.0  
TM                         20.0       12.0  
UA                          0.0        0.0  
US                         10.0        6.0  
UZ                          0.0        0.0

**2.2.** Проанализировать отчёт и сделать выводы по наблюдаемым метрикам.

Таким образом, можно сделать следующие выводы: наибольшее число установок игры в регионе с кодом UA (29556 установок), наименьшее число установок в регионах с кодом BG, DK, ES, TM, US (меньше 5 установок). Также можно сделать вывод, что число платящих пользователей крайне мало. И число пользователей прошедших 10 уровень также крайне мало, по сравнению с числом пользователей прошедщих обучение.
